lib/temporal/client.rb

```diff
@@ -229,25 +229,29 @@ def await_workflow_result(workflow, workflow_id:, run_id: nil, timeout: nil, nam
       execution_options = ExecutionOptions.new(workflow, options, config.default_execution_options)
       max_timeout = Temporal::Connection::GRPC::SERVER_MAX_GET_WORKFLOW_EXECUTION_HISTORY_POLL
       history_response = nil
-      begin
-        history_response = connection.get_workflow_execution_history(
-          namespace: execution_options.namespace,
-          workflow_id: workflow_id,
-          run_id: run_id,
-          wait_for_new_event: true,
-          event_type: :close,
-          timeout: timeout || max_timeout,
-        )
-      rescue GRPC::DeadlineExceeded => e
-        message = if timeout
-          "Timed out after your specified limit of timeout: #{timeout} seconds"
-        else
-          "Timed out after #{max_timeout} seconds, which is the maximum supported amount."
+      while(true) do
+        begin
+          history_response = connection.get_workflow_execution_history(
+            namespace: execution_options.namespace,
+            workflow_id: workflow_id,
+            run_id: run_id,
+            wait_for_new_event: true,
+            event_type: :close,
+            timeout: timeout || max_timeout,
+          )
+        rescue GRPC::DeadlineExceeded => e
+          message = if timeout
+            "Timed out after your specified limit of timeout: #{timeout} seconds"
+          else
+            "Timed out after #{max_timeout} seconds, which is the maximum supported amount."
+          end
+          raise TimeoutError.new(message)
         end
-        raise TimeoutError.new(message)
+        history = Workflow::History.new(history_response.history.events)
+        closed_event = history.events.first
+
+        break unless closed_event.nil?
```

Comment:
Totally a personal nitpick, but framing this in a positive way ("break if x is present") is easier to parse than "break unless x is nil / not present". This is just a personal thing, and the `unless` form might be totally more approachable for other folks.

I know that the suggest doesn't have quite the same semantics as the current code, — `break if closed_event` would break on _any_ non-nil value — but I think since it's coming from gRPC, we should be able to trust that the value we get back has the expected shape (as long as it's not `nil`)?

```suggestion
        break if closed_event
```

Reply:
Done!